File: old/basic/src/main/java/com/dinesh/android/java/fragment/SecondFragment.java
package com.dinesh.android.java.fragment;

import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentResultListener;
import androidx.fragment.app.FragmentTransaction;

import com.dinesh.android.R;

public class SecondFragment extends Fragment {
    private final String TAG = "log_" + getClass().getName().split(getClass().getName().split("\\.")[2] + ".")[1];

    TextView textView;
    EditText et_SecondFragment;
    Button btnFirstFragment, btnThirdFragment;

    String result;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_second,container,false);

        textView = view.findViewById(R.id.textView);
        et_SecondFragment = view.findViewById(R.id.et_SecondFragment);
        btnFirstFragment = view.findViewById(R.id.btnFirstFragment);
        btnThirdFragment = view.findViewById(R.id.btnThirdFragment);

        FragmentTransaction fragmentTransaction = requireActivity().getSupportFragmentManager().beginTransaction();

        btnFirstFragment.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                result = et_SecondFragment.getText().toString();
                Log.e(TAG, "onClick: FirstFragment");
                Toast.makeText(getActivity(), "Hello FirstFragment", Toast.LENGTH_SHORT).show();
                fragmentTransaction.replace(R.id.fragment, new FirstFragment());
                fragmentTransaction.commit();
            }
        });

        btnThirdFragment.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                result = et_SecondFragment.getText().toString();
                Log.e(TAG, "onClick: ThirdFragment");
                Toast.makeText(getActivity(), "Hello ThirdFragment", Toast.LENGTH_SHORT).show();
                fragmentTransaction.replace(R.id.fragment, new ThirdFragment());
                fragmentTransaction.commit();
            }
        });

        getParentFragmentManager().setFragmentResultListener("requestKey", this, new FragmentResultListener() {
            @Override
            public void onFragmentResult(@NonNull String requestKey, @NonNull Bundle bundle) {
                // We use a String here, but any type that can be put in a Bundle is supported
                result = bundle.getString("bundleKey");
                et_SecondFragment.setText(result);
                Log.i(TAG, "onFragmentResult: " + result);
            }
        });


        return view;
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        Log.e(TAG, "onDestroyView: " );
        Bundle bundle = new Bundle();
        bundle.putString("bundleKey", result);
        getParentFragmentManager().setFragmentResult("requestKey", bundle);
    }

}

File: old/basic/src/main/java/com/dinesh/android/kotlin/fragment/SecondFragment.kt
package com.dinesh.android.kotlin.fragment

import android.os.Bundle
import android.util.Log
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.core.os.bundleOf
import androidx.fragment.app.setFragmentResult
import androidx.fragment.app.setFragmentResultListener
import androidx.navigation.fragment.findNavController
import com.dinesh.android.R
import com.dinesh.android.databinding.FragmentSecondBinding


class SecondFragment : Fragment() {
    private val TAG = "log_" + SecondFragment::class.java.name.split(SecondFragment::class.java.name.split(".").toTypedArray()[2] + ".").toTypedArray()[1]
    private var _binding: FragmentSecondBinding? = null
    private val binding get() = _binding!!
    lateinit var result: String

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        _binding = FragmentSecondBinding.inflate(inflater, container, false)
        Log.d(TAG, "onCreateView: ")

        passDataBetweenFragment()

        return binding.root
    }

    private fun passDataBetweenFragment() {
        binding.btnFirstFragment.setOnClickListener {
            result = binding.etSecondFragment.text.toString()
            Log.d(TAG, "btnFirstFragment: ${result}")
            findNavController().navigate(R.id.action_secondFragment_to_firstFragment)
        }

        binding.btnThirdFragment.setOnClickListener {
            result = binding.etSecondFragment.text.toString()
            Log.d(TAG, "btnThirdFragment: ${binding.etSecondFragment.text}")
            findNavController().navigate(R.id.action_secondFragment_to_thirdFragment)
        }

        setFragmentResultListener("requestKey") { requestKey, bundle ->
            result = bundle.getString("bundleKey").toString()
            Log.e(TAG, "setFragmentResultListener: ${result}")
            binding.etSecondFragment.setText(result)
            Toast.makeText(activity, result.toString(), Toast.LENGTH_SHORT).show()
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        Log.e(TAG, "onDestroyView: ")

        //  passDataBetweenFragment
        setFragmentResult("requestKey", bundleOf("bundleKey" to result))

        _binding = null
    }

}